zkevm-circuits/src/evm_circuit/execution/logs.rs

```diff
@@ -10,7 +10,7 @@ use crate::{
                 Transition::{Delta, To},
             },
             memory_gadget::{MemoryAddressGadget, MemoryExpansionGadget},
```

Comment:
I think `Word::from_lo()` is still useful, but we should only use it when we can guarantee that the value is < 128 bits.  I think there are cases where we can guarantee that like in `ComparatorGadget`.

From the current PR, I think `MemoryAddressGadget` should take `Word32` instead of `[Cell<F>; N_BYTES_MEMORY_ADDRESS]` to keep equivalency to the previous code.

Reply:
Have few change
- rename `from_lo()` to `from_lo_unchecked()`
- add TODO for write part range check 

I think we can separate read and write part instead of single function `is_write: bool`. Then in read part we can use `from_lo_unchecked()` safely, and only range check on write part.  If this plan sound good, I will proceed this change soon